themes/default/content/blog/api-gateway-to-eventbridge/index.md

```diff
@@ -0,0 +1,371 @@
+---
+title: "API Gateway to EventBridge with Pulumi"
+date: 2022-01-07
+draft: true
+meta_desc: These are words that add up to at least 50 characters.
+meta_image: meta.png
+authors:
+    - christian-nunciato
+tags:
+    - aws
+    - api-gateway
+    - eventbridge
+    - lambda
+---
+
+If you're familiar with [Amazon API Gateway](https://aws.amazon.com/api-gateway/), you know it's all about making it easier to provision and manage a web API. Maybe you've used it, as I have, with [Crosswalk]({{< relref "/docs/guides/crosswalk/aws" >}}), our AWS extension library, to stand up a REST API and handle requests with AWS Lambda functions:
+
+```typescript
+import * as awsx from "@pulumi/awsx";
+
+// Create a new API Gateway instance.
+const api = new awsx.apigateway.API("my-api", {
+    routes: [
+        {
+            // Define an HTTP endpoint.
+            method: "GET",
+            path: "/things",
+
+            // Handle requests with an AWS Lambda function.
+            eventHandler: async (apiGatewayEvent) => {
+                return {
+                    statusCode: 200,
+                    body: JSON.stringify([
+                        "thingOne",
+                        "thingTwo",
+                    ]),
+                };
+            },
+        },
+    ],
+});
+
+// Export the API's public URL. 🎉
+export const apiUrl = api.url;
+```
+
+I love this abstraction, and I use it all the time; it's an incredibly convenient way to solve a really common problem. But if you look closely at the code, you'll see that it's making a fairly strong assumption about how you'll be handling HTTP requests --- namely, that you'll do so with a single Lambda function, and that that function will always return a JavaScript object [of a particular shape](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-integration-settings-integration-response.html).
```

Comment:
I must admit, I've always been a spaces-around-the-dashes kinda person. I've seen it both ways, and it seems to be one of those things that's left up to the publication to decide. I'm happy to standardize on no-spaces, though, but it'll be a hard habit to break. 😄

Reply:
I'm mostly going off of a number of style guides from my editing years, but for tech docs, I go with https://docs.microsoft.com/en-us/style-guide/punctuation/dashes-hyphens/emes

Reply:
(I also personally like `---` (over `&mdash;`) 'cause quicker to type, generally works, and a bit easier on the eyes.)

Reply:
Yep! I personally don't have any preference (I've always used the HTML code, but that's really just a what-I-was-taught thing), so use whichever you'd like :) If you think it's easier for people to use/remember/read, it might be worth a style guide note.